src/makeRobotAccountant.js

```diff
@@ -6,7 +6,25 @@
  */
 
 function makeRobotAccountant() {
-  // write code here
+  let count = 1;
+
+  const innerFunc = function (x) {
```

Comment:
Let's refactor this a little, step by step:
- no need to declare this function as a variable just to return it later, you can just add return here
```js
function makeRobotAccountant() {
  let count = 1;

  return function (x) {
    if (count <= 3 || (count > 3 && count % 2 !== 0)) {
      return function (y) {
        count++;

        return x + y;
      };
    } else {
      return function () {
        count++;

        return 'Bzzz... Error!';
      };
    }
  };
}
```

Reply:
- you don't need to return two separate functions depending on the condition, just return one handling this condition
```js
function makeRobotAccountant() {
  // write code here
  let count = 1;

  return function (x) {
    return function (y) {
      if (count <= 3 || (count > 3 && count % 2 !== 0)) {
        count++;

        return x + y;
      } else {
        count++;

        return 'Bzzz... Error!';
      }
    };
  };
}
```

Reply:
- now you probably see the unnecessary repetition of the count++, it may be done before if else block as we always have to increment the count regardless of the condition outcome (however we need to start from 0 as the if checks count after incrementation)
```js
function makeRobotAccountant() {
  // write code here
  let count = 0;

  return function (x) {
    return function (y) {
      count++;

      if (count <= 3 || (count > 3 && count % 2 !== 0)) {
        return x + y;
      } else {
        return 'Bzzz... Error!';
      }
    };
  };
}
```

Reply:
- lastly let's simply the condition, swap around the return values and make the function have a default return value regardless of the condition
```js
function makeRobotAccountant() {
  let count = 0;

  return function (x) {
    return function (y) {
      count++;

      if (count > 3 && count % 2 === 0) {
        return 'Bzzz... Error!';
      }

      return x + y;
    };
  };
}
```